src/components/popover/popover.jsx

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React, {
+  forwardRef,
+  useEffect,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+import IconButton from '../icon-button/icon-button.jsx';
+import iconClear from '../../svgs/clear.svg';
+import styles from './popover.css';
+
+const Popover = forwardRef(function Popover(props, ref) {
+  const [open, setOpen] = useState(false);
+  const closeIconRef = useRef();
+  const backupRef = useRef();
+  const selfRef = ref || backupRef;
+
+  const onWindowClick = () => {
+    if (open) {
+      setOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('click', onWindowClick);
+
+    if (closeIconRef.current && open) {
+      closeIconRef.current.focus();
+    }
+
+    if (!open) {
+      props.onClose();
+    }
+
+    return () => {
+      window.removeEventListener('click', onWindowClick);
+    };
+  }, [open]);
+
+  useImperativeHandle(selfRef, () => ({
+    get open() {
+      return open;
+    },
+    set open(value) {
+      setOpen(value);
+    },
+  }));
+
+  if (!open) {
+    return null;
+  }
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
```

Comment:
We don't need to bypass the rule if we provide a list of roles for the node, somewhere in this doc https://www.w3.org/TR/2014/REC-wai-aria-20140320/host_languages#host_general_role

e.g. `role="dialog presentation"`

Reply:
To satisfy the semantics and the linter, I created a new inner `div` with `role="presentation"`, and the event handler is now there instead.

See also: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-noninteractive-element-interactions.md#case-this-element-is-catching-bubbled-events-from-elements-that-it-contains

Reply:
Does adding two roles instead of an additional div make more sense?

Reply:
Unfortunately, two roles doesn't seem to satisfy the linter, and by my reading, isn't what the spec outlines either. 😬

Reply:
Oh, bummer. It's probably not popular to use a list of roles. 

Or I am just misinterpreting:

> The attribute value MUST allow a token list as the value;

Reply:
I must've missed _list_ there, cuz yeah, that does seem like it should work then. 😬 Maybe a mismatch between spec and browser implementation? Or at least between spec and linter...